qcodes_contrib_drivers/drivers/Vaunix/LDA.py

```diff
@@ -0,0 +1,291 @@
+        wf_vals = LdaWorkingFrequency.get_validator(parent)
+        if wf_vals:
+            self.add_parameter("working_frequency",
+                               parameter_class=LdaWorkingFrequency,
+                               vals=wf_vals,
+                               )
+
+
+class LdaParameter(Parameter):
+    scaling = 1.0  # Scaling from integers from API to physical quantities
+
+    def __init__(self, name: str,
+                 instrument: Union[Vaunix_LDA, LdaChannel],
+                 dll_get_function: Callable, dll_set_function: Callable,
+                 **kwargs):
+        """
+        Parameter associated with one channel of the LDA.
+
+        Args:
+            name: parameter name
+            instrument: parent instrument, either LDA or LDA channel
+            dll_get_function: DLL function that gets the value
+            dll_get_function: DLL function that sets the value
+        """
+        super().__init__(name, instrument, **kwargs)
+        self.dll = instrument.root_instrument.dll
+        self.reference = instrument.root_instrument.reference
+        self.dll_get_function = partial(dll_get_function, self.reference)
+        self.dll_set_function = partial(dll_set_function, self.reference)
+
+    def _switch_channel(self) -> None:
+        """
+        Switch to this channel.
+        """
+        if hasattr(self.instrument, "channel_number"):
+            instr = cast(Instrument, self.instrument)
+            self.dll.fnLDA_SetChannel(self.reference,
+                                      instr.channel_number)
+
+    def get_raw(self) -> float:
+        """
+        Switch to this channel and return current value.
+        """
+        self._switch_channel()
+        value = self.dll_get_function()
+        if value < 0:
+            raise RuntimeError(f'{self.dll_get_function.func.__name__} '
+                               f'returned error {value}')
+        return value * self.scaling
+
+    def set_raw(self, value: float) -> None:
+        """
+        Switch to this channel and set to ``value`` .
+        """
+        self._switch_channel()
+        value = round(value / self.scaling)
+        error_msg = self.dll_set_function(value)
+        if error_msg != 0:
+            raise RuntimeError(f'{self.dll_set_function.func.__name__} '
+                               f'returned error {error_msg}')
+
+
+class LdaAttenuation(LdaParameter):
+    """
+    Attenuation of one channel in the LDA.
+    """
+    scaling = 0.05  # integers returned by the API correspond to 0.05 dB
+
+    def __init__(self, name: str,
+                 instrument: Union[Vaunix_LDA, LdaChannel],
+                 **kwargs):
+        dll = instrument.root_instrument.dll
+        super().__init__(name, instrument,
+                         dll_get_function=dll.fnLDA_GetAttenuationHR,
+                         dll_set_function=dll.fnLDA_SetAttenuationHR,
+                         vals=Numbers(0, 63),  # replace below
```

Comment:
can't the validator be prepared before calling `super.__init__`?

Reply:
Fixed in 5d9a29e.